### PR TITLE
kafka_connectors module: fix for issue 401

### LINKED ISSUE #401: Kafka Connect Module Output
Apart from the fact that this is a very great feature and makes life much easier, the output is unfortunately often not very helpful. You get a 400 return when deploying n connectors, no matter if the JDBC password in one connector is not correct or if the format of another connector is not suitable. At least this is my impression so far. I haven't had time to look into the code in more detail, but maybe someone can answer the question, if this is due to the Connector API, or if this could be improved in the module?

### PR DESCRIPTION
**Pull request by ldom**
# Description

revamped error management, status check on the connector tasks

Fixes #401 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested and debugged locally with multiple connectors (using file sink) over python 2 and 3.

**Test Configuration**:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.kafka_connect/library/kafka_connectors.py [hunk 1]
```diff
@@ -62,17 +68,76 @@ def get_current_connectors(connect_url):
...
     url = "{}/{}".format(connect_url, name)
     r = open_url(method='DELETE', url=url)
     return r.getcode() == 200
 
+
+# return value: success (bool), changed (bool), message (str)
 def create_new_connector(connect_url, name, config):
     data = json.dumps({'name': name, 'config': config})
     headers = {'Content-Type': 'application/json'}
     r = open_url(method='POST', url=connect_url, data=data, headers=headers)
```
Comment (jeqo) [inline roles/confluent.kafka_connect/library/kafka_connectors.py:82]: Not 100% sure, but all non-20x responses (e.g. 409) will be reported as exception, and will blow up. 

imho, all calls to `open_url` should be try-catched, as 

https://github.com/confluentinc/cp-ansible/blob/1409fd5e207c14c654aa05eab57748a9ee25b00d/roles/confluent.kafka_connect/library/kafka_connectors.py#L62-L69

fwiw, I've spotted this before and locally fixed some of these: https://github.com/confluentinc/cp-ansible/compare/6.0.0-post...jeqo:fix-401

**Comment by nsharma-git** [comment]
Hello @ldom 
Sorry for long pending action item on this PR. 

Does this PR still hold good ? If thats the case, can you please target this to 6.0.x branch and merge the changes in https://github.com/confluentinc/cp-ansible/compare/6.0.0-post...jeqo:fix-401 . Thanks!

**Comment by nsharma-git** [comment]
Hello @ldom 
Sorry for long pending action item on this PR. 

Does this PR still hold good ? If thats the case, can you please target this to 6.0.x branch and merge the changes in https://github.com/confluentinc/cp-ansible/compare/6.0.0-post...jeqo:fix-401 . Thanks!

**Comment by nsharma-git** [comment]
> Sorry @nsharma-git, just got back. This PR still holds good and I'll change the target branch, but the main issue is that I never got the molecule test environment up properly and I lack time to work on it. However, the fix is definitely useful and actually used at some customer sites.

Thanks @ldom. Will see, if we can help here.

**Comment by wadhwa1** [comment]
Thank you @ldom for your help and contribution! :)
we have now added molecule scenarios on top of this PR + fixed some known issues in kafka_connectors module to support ssl_enabled=true. Here's the PR https://github.com/confluentinc/cp-ansible/pull/1296
Closing this one. thx

**Pull request closed without merging**